src/pyocto/associator.py

```diff
@@ -644,56 +646,67 @@ def associate(
         picks_org["idx"] = np.arange(len(picks_org))
 
         if self.second_pass_overwrites is not None:
-            # Perform second association pass with the unused picks
-            picks_sub = picks_org[~picks_org["idx"].isin(assignments["pick_idx"])].copy(
-                deep=True
-            )  # Only unused picks
-            picks_sub["idx2"] = np.arange(len(picks_sub))
-            picks_cpp = self._convert_picks(picks_sub)
-
-            config = self._build_config(stations_cpp, second_pass=True)
-
-            associator = backend.OctoAssociator(config)
-
-            events_cpp = associator.associate(picks_cpp)
-            self._cached_pointers = (
-                {}
-            )  # Delete cached objects and allow garbage collection
-            events2, assignments2 = self._parse_events(events_cpp)
-
-            if len(events2) > 0:  # Skip this if there is anyhow no event
-                # Update event_idx to be non-intersecting with original output
-                if len(events) > 0:
-                    event_idx_correction = events["idx"].max() + 1
-                    events2["idx"] += event_idx_correction
-                    assignments2["event_idx"] += event_idx_correction
-
-                # Map pick idx to the orginal one (idx) instead of the subset one (idx2)
-                assignments2 = pd.merge(
-                    assignments2,
-                    picks_sub,
-                    left_on="pick_idx",
-                    right_on="idx2",
-                    validate="m:1",
-                )
-                assignments2.drop(columns="pick_idx", inplace=True)
-                assignments2.rename(columns={"idx": "pick_idx"}, inplace=True)
-                assignments2 = assignments2[
-                    ["event_idx", "pick_idx", "residual"]
-                ].copy()
-
-                events = pd.concat([events, events2])
-                assignments = pd.concat([assignments, assignments2])
-
-                events.sort_values("time", inplace=True)
-                events.reset_index(drop=True, inplace=True)
-                assignments.reset_index(drop=True, inplace=True)
+            second_pass_iterations = self.second_pass_overwrites.pop("iterations",
+                                                                     None)
+            if not second_pass_iterations:
```

Comment:
You can just put 1 as the default value in the `pop` before instead of the `None`.